src/rng/src/main.rs

```diff
@@ -0,0 +1,192 @@
+// VIRTIO RNG Emulation via vhost-user
+//
+// Copyright 2021 Linaro Ltd. All Rights Reserved.
+// Mathieu Poirier <mathieu.poirier@linaro.org>
+//
+// SPDX-License-Identifier: Apache-2.0
+
+mod vhu_rng;
+
+use clap::{load_yaml, App};
+use std::fs::File;
+use std::path::Path;
+use std::sync::{Arc, Mutex, RwLock};
+use std::thread;
+use vhost::{vhost_user, vhost_user::Listener};
+use vhost_user_backend::VhostUserDaemon;
+use vhu_rng::VuRngBackend;
+use vm_memory::{GuestMemoryAtomic, GuestMemoryMmap};
+
+fn start_backend(cmd_args: clap::ArgMatches, dry_run: bool) -> Result<(), String> {
+    let mut handles = Vec::new();
+
+    let path = cmd_args
+        .value_of("socket_path")
+        .ok_or("Invalid socket path")?;
+
+    let count = cmd_args
+        .value_of("socket_count")
+        .unwrap_or("1")
+        .parse::<u32>()
+        .map_err(|_| "Invalid socket_count")?;
+
+    let source = match cmd_args.value_of("rng_source") {
+        Some(source) => {
+            if Path::new(source).is_file() {
+                source.to_string()
+            } else {
+                return Err(String::from("Enable to access RNG source file"));
+            }
+        }
+        None => "/dev/urandom".to_string(),
+    };
+
+    let random_file = match File::open(source) {
+        Ok(random_file) => Arc::new(Mutex::new(random_file)),
+        Err(e) => return Err(e.to_string()),
+    };
+
+    let period_ms: u128 = match cmd_args.value_of("period") {
+        Some(period_ms) => match period_ms.parse::<u128>() {
+            Ok(value) => {
+                if value > (1 << 16) {
```

Comment:
What is `1 << 16`? Is this magic number defined in a specification? Can we define a `const` for it and add some links to where it is defined?

Reply:
It is simply the value 65536, chosen to match the value of the virtio RNG device implementation in QEMU.  I will make a constant and add a comment.